utilities/profilegenerator.py

```diff
@@ -0,0 +1,75 @@
+from components.chassis import Chassis
+import math
+
+def generate_interpolation_trajectory(x_start, x_final, v_max):
+    """Generate a 1d interpolation profile, where the velocity is constant
+    over the duration of the trajectory.
+
+    :returns: a list of (pos, vel acc) tuples.
+    """
+
+    x = x_final - x_start
+    num_segments = int(x/v_max*Chassis.motion_profile_speed)
+    segments = [(i/num_segments, v_max, 0) for i in range(0, num_segments+1)]
+    return segments
+
+def generate_trapezoidal_trajectory(
+        x_start, v_start, x_final, v_final, v_max, a_pos, a_neg):
+    """ Generate a 1d trapezoidal profile.
+
+    :returns: a list of (pos, vel acc) tuples
+    """
+
+    # area under the velocity-time trapezoid
+    x = x_final - x_start
+
+    # find the max reachable velocity if we spend all our time accelerating
+    # and decelerating. Used as max velocity in cases where we don't hit the
+    # robot's top speed
+    triangular_max = math.sqrt(
+            (2*x*a_pos*a_neg+a_neg*v_start**2+a_pos*v_final**2)/(a_neg-a_pos))
```

Comment:
Hmm. Is there a scenario where we would want to have `a_neg != -a_pos`? If we constrain `a_neg == -a_pos`, we could simplify this expression a _lot_.